modules/ivc_champva/app/services/ivc_champva/attachments.rb

```diff
@@ -6,31 +6,49 @@ module Attachments
 
     def handle_attachments(file_path) # rubocop:disable Metrics/MethodLength
       file_paths = [file_path]
+      file_processing_errors = []
+
+      Rails.logger.info('IVC ChampVA Forms - Getting attachments')
       attachments = get_attachments
 
+      Rails.logger.info("IVC ChampVA Forms - Processing #{attachments.size} attachments for form #{form_id} #{uuid}")
       attachments.each_with_index do |attachment, index|
         new_file_name = if attachment.include?('_additional_')
+                          Rails.logger.info("IVC ChampVA Forms - Attachment #{index} is additional")
                           "#{uuid}_#{File.basename(attachment,
                                                    '.*')}.pdf"
                         else
+                          Rails.logger.info("IVC ChampVA Forms - Attachment #{index} is supporting")
                           "#{uuid}_#{form_id}_supporting_doc-#{index}.pdf"
                         end
 
         new_file_path = if Flipper.enabled?(:champva_pdf_decrypt, @current_user)
+                          Rails.logger.info("IVC ChampVA Forms - Attachment #{index} is using decrypt path")
                           File.join('tmp', new_file_name)
                         else
+                          Rails.logger.info("IVC ChampVA Forms - Attachment #{index} is using original path")
                           File.join(File.dirname(attachment), new_file_name)
                         end
 
         if Flipper.enabled?(:champva_pdf_decrypt, @current_user)
           # Use FileUtils.mv to handle `Errno::EXDEV` error since encrypted PDFs
           # get stashed in the clamav_tmp dir which is a different device on staging, apparently
+          Rails.logger.info("IVC ChampVA Forms - Handling attachment #{index} via mv")
           FileUtils.mv(attachment, new_file_path) # Performs a copy automatically if mv fails
         else
+          Rails.logger.info("IVC ChampVA Forms - Handling attachment #{index} via rename")
           File.rename(attachment, new_file_path)
         end
 
         file_paths << new_file_path
+      rescue => e
+        file_processing_errors << "Error processing attachment at index #{index}: #{e.message}"
+      end
+
+      unless file_processing_errors.empty?
+        error_message = "Unable to process all attachments: #{file_processing_errors.join(', ')}"
```

Comment:
Is there any PII that could potentially be outputted here in the file_processing_errors? I know sometimes filenames can contain user input (potentially PII)

Reply:
Good catch!  I added some special handling for filesystem related errors that avoids passing on any info that may contain user input.

Reply:
Thank you!